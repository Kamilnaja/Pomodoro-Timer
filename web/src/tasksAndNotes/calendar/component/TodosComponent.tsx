export const TodosComponent = () => <div>Todos</div>;
