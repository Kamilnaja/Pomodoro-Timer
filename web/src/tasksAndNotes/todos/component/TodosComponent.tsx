import React from 'react';
import { Task } from '../../../../../types/tasksAndNotesInterfaces';
import { CardContainer } from '../../shared/card/container/CardContainer';

export const TodosComponent = (props: { todos: Task[]; isAddingTaskActive: boolean }) => (
  <div>
    <h2>Your things to do</h2>
    {props.isAddingTaskActive && <CardContainer></CardContainer>}

    {!props.todos.length && <div>You have 0 todos, please add something</div>}
    <ul className="list-unstyled">
      {props.todos?.map((item: Task, idx: number) => (
        <li key={idx} className="mb-1" tabIndex={0}>
          <CardContainer task={item}></CardContainer>
        </li>
      ))}
    </ul>
    <hr />
    <h2>Done tasks</h2>
    <ul className="list-unstyled">
      {props.todos
        .filter(item => item.isDone)
        .map((item: Task, idx: number) => (
          <li key={idx} className="mb-4">
            <CardContainer task={item}></CardContainer>
          </li>
        ))}
      <li></li>
    </ul>
  </div>
);
