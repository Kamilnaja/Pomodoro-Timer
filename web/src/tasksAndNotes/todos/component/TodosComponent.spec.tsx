import { shallow } from 'enzyme';
import toJson, { shallowToJson } from 'enzyme-to-json';
import { ReactComponentElement } from 'react';
import { createTasks } from '../../testing/taskAndNotes.test.data';
import { TodosComponent } from './TodosComponent';

describe('TodosComponent', () => {
  let wrapper = shallow(<TodosComponent todos={createTasks()} isAddingTaskActive={true} />);

  it('Should match snapshot', () => {
    expect(shallowToJson(wrapper)).toMatchSnapshot();
  });
});
