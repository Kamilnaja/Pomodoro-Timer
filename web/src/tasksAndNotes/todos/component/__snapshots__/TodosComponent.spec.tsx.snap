// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TodosComponent Should match snapshot 1`] = `
<div>
  <h2>
    Your things to do
  </h2>
  <Connect(CardContainer) />
  <ul
    className="list-unstyled"
  >
    <li
      className="mb-1"
      key="0"
      tabIndex={0}
    >
      <Connect(CardContainer)
        task={
          Object {
            "dateCreated": 2021-01-11T23:00:00.000Z,
            "dueDate": 2020-01-11T23:00:00.000Z,
            "id": "1",
            "isDone": false,
            "note": "lorem ipsum dolor",
            "subtasks": Array [],
            "title": "hello world",
            "type": 1,
          }
        }
      />
    </li>
  </ul>
  <hr />
  <Accordion>
    <h2>
      Done tasks
      <ForwardRef
        as={
          Object {
            "$$typeof": Symbol(react.forward_ref),
            "defaultProps": Object {
              "active": false,
              "disabled": false,
              "variant": "primary",
            },
            "render": [Function],
          }
        }
        eventKey="0"
        variant="link"
      >
        exp
      </ForwardRef>
    </h2>
    <AccordionCollapse
      eventKey="0"
    >
      <ul
        className="list-unstyled"
      >
        <li />
      </ul>
    </AccordionCollapse>
  </Accordion>
</div>
`;
